earthfile2llb/converter.go

```diff
@@ -1140,7 +1140,20 @@ func (c *Converter) Expose(ctx context.Context, ports []string) error {
 	}
 	c.nonSaveCommand()
 	for _, port := range ports {
-		c.mts.Final.MainImage.Config.ExposedPorts[port] = struct{}{}
+		var exposedPort string
+		if strings.Contains(port, ":") {
+			port = strings.SplitN(port, ":", 2)[1]
+		}
+		if strings.Contains(port, "/") {
+			protocol := strings.SplitN(port, "/", 2)[1]
+			if protocol != "tcp" && protocol != "udp" {
+				return errors.New("Invalid protocol provided")
```

Comment:
It's best to move all of this parsing logic into interpreter.go, and then you can make use of 
```
			return i.errorf(err, cmd.SourceLocation, "failed to parse EXPOSE %s: invalid protocol %s", port, protocol)
```

which will then display the corresponding Earthfile source location.

Reply:
fixed